Duke.java

```diff
@@ -0,0 +1,185 @@
+import java.util.Scanner;
+
+public class Duke {
+    // static variables
+    private static Task[] taskList = new Task[100];
+    private static int taskCounter = 0;
+
+    static String HORIZONTAL_LINE = "------------------------------------------------------------";
+
+    // class methods
+    static void errorMessage() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("I don't know what that means :(");
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    static boolean checkSpace(String line) {
+        // in the case of "todo"
+        if (!line.contains(" ")) {
+            return false;
+        }
+
+        // in the case of "todo "
+        int spaceLocation = line.indexOf(" ");
+        if (spaceLocation == line.length()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static void main(String[] args) {
+        greet();
+        Scanner input = new Scanner(System.in);
+        String line = input.nextLine();
+        String command = line.split(" ")[0];
+
+        while (!command.matches("bye")) {
+            switch (command) {
+            case "list":
+                listTasks();
+                break;
+            case "mark":
+                int index = Integer.parseInt(line.split(" ")[1]) - 1;
+                markTask(taskList[index]);
```

Comment:
These two statements have different levels of abstraction so consider further extracting the logic (SLAP more) into another method to make the code more readable.